gps_ubxm8_i2c/gps.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <i2c.h>
+
+#include "ubxMessages.h"
+#include "ubxPacket.h"
+
+/**
+ * Class for interfacing over i2c with any M8 GPS.
+ * Before this class is used, the GPS reset pin needs to be pulled high.
+ *
+ * This class keeps its own finite state machine for abstracting away polling data availability and retrying requests.
+ * An example usage is as follows:
+ * ```
+ *     GPS myGPS;
+ *     while(1) {
+ *        if(myGPS.pollUpdate(&i2c1) == GPS::PollResult::POLL_FINISHED) {
+ *            UBX_NAV_PVT_PAYLOAD pvtData = myGPS.getSolution();
+ *            myGPS.reset();
+ *        }
+ *     }
+ * ```
+ *
+ * After a packet is received, you _need_ to call the reset method.
```

Comment:
Why have an `Init()` method instead of just putting that functionality in the constructor?

Reply:
The instance of the object is defined in the global scope, hence the constructor is executed before `main`. If the constructor contains code that requires interfacing with hardware - it will error out because the hardware isn't configured yet